File: src/main/resources/static/core/article/article.service.js
'use strict';

angular.module('core.article')
	.service('ArticleService', function($http) {
		this.getAll = () => {
			return $http.get('/api/articles/');
		};

		this.getPdf = (articleId) => {
			return $http.get(`/api/articles/${articleId}`);
		};

		this.searchAll = (query) => {
			return $http.get(`/api/articles/search/${query}`);
		};

		this.searchAdvanced = (query) => {
			return $http.put('/api/articles/search/', query);
		};

		this.searchMoreLikeThis = (articleId) => {
			return $http.get(`/api/articles/search/more-like-this/${articleId}`);
		};

		this.searchGeodistance = () => {
			return $http.get('/api/articles/search/geodistance');
		};
	});

File: src/main/java/ftnjps/scientificcenter/article/ArticleService.java
package ftnjps.scientificcenter.article;

import java.util.List;
import java.util.Map;

import ftnjps.scientificcenter.users.ApplicationUser;

public interface ArticleService {

	Article findOne(Long id);

	List<ArticleDto> findAll(ApplicationUser payer);

	String getPdf(Long id, ApplicationUser forUser);

	Article add(Article article);

	void createIndex();
	boolean index(Article article);
	List<ArticleDto> searchAll(String query, ApplicationUser payer);
	List<ArticleDto> searchAdvanced(Map<String, Object> query, ApplicationUser payer);
	List<ArticleDto> searchMoreLikeThis(Long articleId, ApplicationUser payer);
	List<ArticleDto> searchGeodistance(String location, ApplicationUser payer);

}
